### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by Sumned**

### CODE REVIEW

File: src/main/java/core/basesyntax/BooleanExpression.java [hunk 1]
```diff
@@ -14,6 +14,7 @@
      * без использования конструкций if-else</p>
      */
     public boolean booleanExpression(boolean a, boolean b, boolean c, boolean d) {
-        return false;
+        return (a && b && !c && !d) || (!a && !b && c && d) || (!a && b && c && !d)
+                || (a && !b && !c && d) || (a && !b && c && !d) || (!a && b && !c && d);
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/BooleanExpression.java]: try to simplify it with binary operators

Comment (Sumned) [reply src/main/java/core/basesyntax/BooleanExpression.java]: do you mean that I can make the second part look like the first one, but using "!"?

File: src/main/java/core/basesyntax/BooleanExpression.java [hunk 2]
```diff
@@ -14,6 +14,6 @@
      * без использования конструкций if-else</p>
      */
     public boolean booleanExpression(boolean a, boolean b, boolean c, boolean d) {
-        return false;
+        return ((a != c && b != d)) || ((a != d && b != c));
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/BooleanExpression.java]: try to solve this with XOR(google it)

File: src/main/java/core/basesyntax/BooleanExpression.java [hunk 3]
```diff
@@ -14,6 +14,6 @@
      * без использования конструкций if-else</p>
      */
     public boolean booleanExpression(boolean a, boolean b, boolean c, boolean d) {
-        return false;
+        return ((a ^ c && b ^ d)) || ((a ^ d && b ^ c));
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/BooleanExpression.java]: do we need all  these brackets?

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request closed without merging**